lib/dialyzer/src/dialyzer_dataflow.erl

```diff
@@ -642,8 +641,8 @@ get_apply_fail_msg(Fun, Args, ArgTypes, NewArgTypes,
 	 dialyzer_contracts:contract_to_string(C)};
       none -> {false, none}
     end,
-  EnumArgTypes = lists:zip(lists:seq(1, length(NewArgTypes)), NewArgTypes),
-  ArgNs = [Arg || {Arg, Type} <- EnumArgTypes, t_is_none(Type)],
+  ArgNs = [Arg || Arg <- lists:seq(1, length(NewArgTypes)) &&
```

Comment:
Nitpick:
```suggestion
  ArgNs = [Arg || Arg <- lists:enumerate(NewArgTypes) &&
```

Reply:
I think you mean:

```
  ArgNs = [Arg || {Arg,Type} <:- lists:enumerate(NewArgTypes),
                  t_is_none(Type)],
```

That works but produces more garbage.